src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggerPass.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Bundle\FrameworkBundle\Logger\Logger;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Registers the default logger if necessary.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class LoggerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $loggerDefinition = $container->register('logger', Logger::class);
+        $loggerDefinition->setPublic(false);
+        if ($container->getParameter('kernel.debug')) {
```

Comment:
How does that work with the console. Wouldn't we have too much output now?

Reply:
I'm not sure of what the best behavior is.

Imagine the following scenario:
* I've a problem in production
* I update the env var `APP_DEBUG` to `true`

I want all the available level of logs to be able to debug. (the current behavior).

However, Symfony is very verbose, `APP_DEBUG` is set to `true` for the `dev` env, and `dev` is the default env when the framework is installed. I agree that having a bunch of useless logs in the console when just running `make serve` is not a good developer experience.

What do you think about providing a custom logger definition (with less output) in the Flex recipe for just the `dev` env? We can also disable logs in the `test` env.